src/exotic-sizes.md

```diff
@@ -125,9 +153,30 @@ But this trick doesn't work yet.
 
 One final subtle detail about empty types is that raw pointers to them are
 actually valid to construct, but dereferencing them is Undefined Behavior
-because that doesn't actually make sense. That is, you could model C's `void *`
-type with `*const Void`, but this doesn't necessarily gain anything over using
-e.g. `*const ()`, which *is* safe to randomly dereference.
+because that wouldn't make sense.
+
+We recommend against modelling C's `void*` type with `*const Void`.
+A lot of people started doing that but quickly ran into trouble because
+people want to convert raw pointers to references when doing FFI, and making an
+`&Void` is Undefined Behaviour. `*const ()` (or equivalent) works just as well,
+and can be made into a reference without any safety problems. The only downside
+of modeling `void*` with `*const ()` is that attempts to read or write values
+will silently succeed (doing nothing), instead of producing a compiler error.
```

Comment:
Reading an `*const Void` also doesn't necessarily produce an error -- instead of nothing, you just got UB.